### PR TITLE
[Dubbo-6963] fix bug: delete last config not take effect

### LINKED ISSUE #6963: 最后一条动态配置删除不生效
**Dubbo version**: 2.7.6

在动态配置只剩下一条时，删除会不生效。
影响代码在RegistryProtocol的内部类OverrideListener的doOverrideIfNecessary方法中。
````
//The current, may have been merged many times
URL currentUrl = exporter.getInvoker().getUrl();
//Merged with this configuration
URL newUrl = getConfigedInvokerUrl(configurators, currentUrl);
newUrl = getConfigedInvokerUrl(providerConfigurationListener.getConfigurators(), newUrl);
newUrl = getConfigedInvokerUrl(serviceConfigurationListeners.get(originUrl.getServiceKey())
        .getConfigurators(), newUrl);
if (!currentUrl.equals(newUrl)) {
    RegistryProtocol.this.reExport(originInvoker, newUrl);
    logger.info("exported provider url changed, origin url: " + originUrl +
            ", old export url: " + currentUrl + ", new export url: " + newUrl);
}
````

代码里面用configurators来修改currentUrl，生成一个newUrl，并和currentUrl做对比，如果不一致则重新暴露。

考虑删除最后一条configurator，也就是configurators字段为空的情况，最后newUrl会等于currentUrl，if (!currentUrl.equals(newUrl))结果为false，删除不生效。

应该用configurators修改originUrl，最后和currentUrl做匹配，改为：
````
URL newUrl = getConfigedInvokerUrl(configurators, originUrl);
````

### PR DESCRIPTION
**Pull request by wikiwikiwiki**
fix #6963 ，修复删除最后一条动态配置不会生效的问题

### CODE REVIEW

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/RegistryProtocol.java [hunk 1]
```diff
@@ -653,7 +653,7 @@ public synchronized void doOverrideIfNecessary() {
             //The current, may have been merged many times
             URL currentUrl = exporter.getInvoker().getUrl();
             //Merged with this configuration
-            URL newUrl = getConfigedInvokerUrl(configurators, currentUrl);
+            URL newUrl = getConfigedInvokerUrl(configurators, originUrl);
```
Comment (chickenlj) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/RegistryProtocol.java:656]: LGTM.

**Comment by AlbumenJ** [comment]
@wikiwikiwiki Hi, thanks for your contribution.

Please merge the latest master branch to resolve confilcts.

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/RegistryProtocol.java (hunk 1 above)
Comment (wikiwikiwiki) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/RegistryProtocol.java]: > @wikiwikiwiki Hi, thanks for your contribution.
> 
> Please merge the latest master branch to resolve confilcts.

already resolved

**Comment by wikiwikiwiki** [comment]
> @wikiwikiwiki Hi, thanks for your contribution.
> 
> Please merge the latest master branch to resolve confilcts.

already resolved

**Comment by AlbumenJ** [comment]
LGTM.

@wikiwikiwiki Do you have any time to add some unit test cases for this? I will merge this pr later, and we can do it in a new pull request.

**Comment by wikiwikiwiki** [comment]
> LGTM.
> 
> @wikiwikiwiki Do you have any time to add some unit test cases for this? I will merge this pr later, and we can do it in a new pull request.

ok

**Pull request merged**